CodeSniffer/Standards/Generic/Sniffs/PHP/SyntaxSniff.php

```diff
@@ -56,7 +56,7 @@ public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
     {
         $phpPath = PHP_CodeSniffer::getConfigData('php_path');
         if ($phpPath === null) {
-            return;
+            $phpPath = PHP_BINARY;
```

Comment:
A few options I see:

* PHP 5.3 has been unsupported for over a year, so we just assume PHP_BINARY is available.
* Wrap this in a check for the constant, and fall back to the old behaviour if it's unset.
* Use PHP_BINDIR which is in 5.3 and assume the binary is called 'php'.